scheds/rust/scx_layered/src/main.rs

```diff
@@ -1124,8 +1162,51 @@ fn layer_core_order(
             }
         }
         LayerGrowthAlgo::Linear => {
-            for i in 0..topo.cores().len() {
-                core_order.push(i);
+            linear();
+        }
+        LayerGrowthAlgo::Topo => {
+            let spec_nodes = spec.nodes();
+            let spec_llcs = spec.llcs();
+            let topo_nodes = topo.nodes();
+
+            if spec_nodes.len() + spec_llcs.len() == 0 {
+                // XXX: fallback to something more sane (round robin when it exists)
+                linear();
+            } else {
+                let mut offset;
+                for spec_llc in &spec_llcs {
+                    offset = 0;
+                    for topo_node in topo_nodes {
+                        for topo_llc in topo_node.llcs().values() {
+                            if *spec_llc != topo_llc.id() {
+                                continue;
+                            }
+                            for (id, _core) in topo_llc.cores().keys().enumerate() {
+                                let core_id = id + offset;
+                                if !core_order.contains(&core_id) {
+                                    core_order.push(core_id);
+                                }
+                            }
+                        }
+                        offset += topo_node.cores().len();
+                    }
+                }
+                for spec_node in &spec_nodes {
+                    offset = 0;
+                    for topo_node in topo_nodes {
+                        if *spec_node != topo_node.id() {
+                            offset += topo_node.cores().len();
+                            continue;
+                        }
+                        for (id, _core) in topo_node.cores().keys().enumerate() {
```

Comment:
I don't think this is right because the core ids aren't useful and `enumerating` doesn't use the correct id:
```
$ cat /sys/devices/system/cpu/cpu*/topology/core_id  | sort -n| uniq -c | sort -n
      4 0
      4 1
      4 10
      4 11
      4 12
      4 16
      4 17
      4 18
      4 19
      4 2
      4 20
      4 24
      4 25
      4 26
      4 27
      4 28
      4 3
      4 4
      4 8
      4 9
```

Reply:
I don't think this code does what it's intended to do. `BTreeMap::keys(&self)` returns an iterator over the key values, sorted. Calling `.enumerate()` on that returns an iterator of tuples of the index of the sorted key and the key - that doesn't seem very useful to me. I would expect to want to iterate `.keys()` for the IDs or `map.iter()` for a tuple of the keys and values in the map.